[octomap] build failure

### Operating system

Windows

### Compiler

MSVC Visual Studio 2017 15.6

### Steps to reproduce the behavior

```Shell
current octomap version doesn't compile on latset version of Visual Studio 2017 (15.6).

This is fixed in the following (closed) PR:

https://github.com/OctoMap/octomap/pull/398

I think the port should be patched with this fix
```


### Failure logs

```
1>C:\workspace\cranebit3\vcpkg_installed\x64-windows-static-md-v143\include\octomap\ScanGraph.h(91,17): error C2666: 'octomap::ScanNode::operator ==': overloaded functions have similar conversions
1>C:\workspace\cranebit3\vcpkg_installed\x64-windows-static-md-v143\include\octomap\ScanGraph.h(63,10): message : could be 'bool octomap::ScanNode::operator ==(const octomap::ScanNode &)'
1>C:\workspace\cranebit3\vcpkg_installed\x64-windows-static-md-v143\include\octomap\ScanGraph.h(63,10): message : or 'bool octomap::ScanNode::operator ==(const octomap::ScanNode &)' [synthesized expression 'y == x']
1>C:\workspace\cranebit3\vcpkg_installed\x64-windows-static-md-v143\include\octomap\ScanGraph.h(91,17): message : while trying to match the argument list '(octomap::ScanNode, octomap::ScanNode)'
1>C:\workspace\cranebit3\vcpkg_installed\x64-windows-static-md-v143\include\octomap\ScanGraph.h(91,49): error C2666: 'octomap::ScanNode::operator ==': overloaded functions have similar conversions
1>C:\workspace\cranebit3\vcpkg_installed\x64-windows-static-md-v143\include\octomap\ScanGraph.h(63,10): message : could be 'bool octomap::ScanNode::operator ==(const octomap::ScanNode &)'
1>C:\workspace\cranebit3\vcpkg_installed\x64-windows-static-md-v143\include\octomap\ScanGraph.h(63,10): message : or 'bool octomap::ScanNode::operator ==(const octomap::ScanNode &)' [synthesized expression 'y == x']
1>C:\workspace\cranebit3\vcpkg_installed\x64-windows-static-md-v143\include\octomap\ScanGraph.h(91,49): message : while trying to match the argument list '(octomap::ScanNode, octomap::ScanNode)'
```

### Additional context

_No response_

--- a/ports/octomap/fix-compilation.patch
+++ b/ports/octomap/fix-compilation.patch
@@ -1,0 +1,22 @@
+diff --git a/octomap/include/octomap/ScanGraph.h b/octomap/include/octomap/ScanGraph.h
+index 07c7436..e54eaf5 100644
+--- a/octomap/include/octomap/ScanGraph.h
++++ b/octomap/include/octomap/ScanGraph.h
+@@ -60,7 +60,7 @@ namespace octomap {
+ 
+     ~ScanNode();
+ 
+-    bool operator == (const ScanNode& other) {
++    bool operator == (const ScanNode& other) const {
+       return (id == other.id);
+     }
+ 
+@@ -87,7 +87,7 @@ namespace octomap {
+       : first(_first), second(_second), constraint(_constraint), weight(1.0) { }
+     ScanEdge() {}
+ 
+-    bool operator == (const ScanEdge& other) {
++    bool operator == (const ScanEdge& other) const {
+       return ( (*first == *(other.first) ) && ( *second == *(other.second) ) );
+     }
+ 

--- a/ports/octomap/portfile.cmake
+++ b/ports/octomap/portfile.cmake
@@ -7,7 +7,8 @@ vcpkg_from_github(
     SHA512 ec321a5355091acbd3d3fda7c858e2078c29195e73461c8a34db2c4614c6b2e38b35a59671f1071f7eb397cac4df78869f14a13af2e68d64e5a2e2d8727846cd
     HEAD_REF master
     PATCHES
-      "001-fix-exported-targets.patch"
+      001-fix-exported-targets.patch
+      fix-compilation.patch
 )
 
 vcpkg_cmake_configure(
@@ -38,7 +39,7 @@ endif()
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/octomap")
 
 # Handle copyright
-file(INSTALL "${SOURCE_PATH}/octomap/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/octomap/LICENSE.txt")
 
 vcpkg_copy_pdbs()
 

--- a/ports/octomap/vcpkg.json
+++ b/ports/octomap/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "octomap",
   "version": "1.9.6",
+  "port-version": 1,
   "description": "An Efficient Probabilistic 3D Mapping Framework Based on Octrees",
   "homepage": "https://octomap.github.io/",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5922,7 +5922,7 @@
     },
     "octomap": {
       "baseline": "1.9.6",
-      "port-version": 0
+      "port-version": 1
     },
     "ode": {
       "baseline": "0.16.3",

--- a/versions/o-/octomap.json
+++ b/versions/o-/octomap.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b778b0aef20fee697a18865995fff021501e8509",
+      "version": "1.9.6",
+      "port-version": 1
+    },
+    {
       "git-tree": "b6146ecee8dcf9ba6573d11f26c1574e89c546d8",
       "version": "1.9.6",
       "port-version": 0

